modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OAuthConnectionProviderModelValidator.java

```diff
@@ -46,27 +53,62 @@ protected void onConnectionProvider(ConnectionProviderModel model) {
           return;
         }
 
-        AuthorizationCode authCode = implementingType.getAnnotation(AuthorizationCode.class);
-        ClientCredentials clientCredentials = implementingType.getAnnotation(ClientCredentials.class);
+        Optional<OAuthModelProperty> optionalOAuthModelProperty = model.getModelProperty(OAuthModelProperty.class);
 
-        if (authCode != null && clientCredentials != null) {
+        boolean supportsAuthCode = optionalOAuthModelProperty.map(oAuthModelProperty -> oAuthModelProperty.getGrantTypes()
```

Comment:
maybe this code could be extracted to a function and used to calculate supportsAuthCode and supportsClientCredentials? that might reduce the whole method length increasing readability.

Reply:
👍